### ISSUE TITLE
Does recursive container creation leak information?

### ISSUE THREAD

**Issue by RubenVerborgh** [solid/specification#129]

**Comment by michielbdejong** [comment]
If you create `C/R` and `C/` gets created on the fly (transparently), then the information about whether `C/` existed beforehand stays more hidden, not less hidden?

**Comment by csarven** [comment]
Reopening the issue as it was closed by a non-CG member at the time, without CG's consent. The last useful revision in https://github.com/solid/specification/issues/139#issue-550053740 is below, before the comment was erased.

---

This is a security follow-up on #68.

Servers are allowed to use 404 for 2 cases:
1. when a resource does not exist
2. when it does not want to acknowledge whether a resource exists

In #68, the current resolution is to automatically create intermediate containers as needed when a deeper document is created through `PATCH` (and perhaps other methods). In that case, success or failure of that operation might leak information in case 2 above.

**Comment by csarven** [comment]
GitHub's ToS allow users to reference and reuse publicly available comments within GitHub ([see Section D.5](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#5-license-grant-to-other-users)). And, the [Solid GitHub org](https://github.com/solid/) operates under an MIT license, promoting open collaboration and transparency. So, I see no basis for a copyright violation claim.

**Comment by RubenVerborgh** [comment]
The contributions were not publicly available; copyright claims will be filed.
